harvester/harvest.py

```diff
@@ -439,20 +451,24 @@ def validate(self) -> None:
             )
 
     def create_record(self):
-        ckan.action.package_create(**self.ckanified_metadata)
+        result = ckan.action.package_create(**self.ckanified_metadata)
+        self.ckanid = result['id']
 
     def update_record(self) -> dict:
         ckan.action.package_update(**self.ckanified_metadata)
```

Comment:
did you test `package_update`? I believe `ckanified_metadata` method needs to be updated to include that field as `id`.